gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjureRunnerResource.java

```diff
@@ -179,12 +180,13 @@ public void invoke(Project project, String failedTo, List<String> unloggedArgs,
         }
 
         @Override
-        public void invoke(Project project, String failedTo, List<String> unloggedArgs, List<String> loggedArgs) {
-            project.getLogger().info("Running in-process java with args: {}", loggedArgs);
+        public void invoke(
+                ExecOperations _execOperations, String failedTo, List<String> unloggedArgs, List<String> loggedArgs) {
             List<String> combinedArgs = ImmutableList.<String>builder()
                     .addAll(unloggedArgs)
                     .addAll(loggedArgs)
                     .build();
+            log.info("Running in-process java with args: {}", loggedArgs);
```

Comment:
I'm trying to work out _why_ the metaspace was running out and how we expect running in an external worker daemon to help.

I've read this code and it seems we are reverse engineering the classpath from the conjure generator binary, then running the main method. My guess is that we load the classes multiple times per conjure generator invocation (possible over several gradle runs if the gradle daemon is reused) these classes are not released from memory (?) causing it to run out metaspace.

I suspect that the worker daemons are going to have the same issue here, since we repeatedly classload inside of them and the worker daemons are persisted across gradle invocations, and will also OOM. I'm not sure if this will fail the build or whether gradle will try and run it again, but it feels like we aren't fixing the source of the issue - the memory leak.

I guess this is what #898 is trying to do - reuse the classloader rather than make a new one each time.

I think for both #898 and this issue we might be able to kill two birds with one stone - perhaps we can get gradle to manage the lifecycle of the classpath through the `classloaderIsolation` part of the worker api? Rather than spinning out a process isolated worker daemon that has a bit more metaspace overhead and still does our own classloading?

Reply:
Hmm, I see [we use some bytebuddy to replace the `System.exit` method](https://github.com/palantir/gradle-conjure/blob/ed6aa328cb527399c2e4ade472140dcf2f373c64/gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjureRunnerResource.java#L105-L116). @carterkozak do you think it would be possible to still do that with gradle doing the classloading? (and does the above make sense to you?)

Reply:
Actually, I see there's a `toJar` method instead of [loading it directly](https://github.com/palantir/gradle-conjure/blob/ed6aa328cb527399c2e4ade472140dcf2f373c64/gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjureRunnerResource.java#L115-L116), so I'm going to guess yes we can still use gradle for this.

Reply:
That's worth a shot, but I imagine it will be expensive so we'll want to reuse the worker initialization as much as possible.

Reply:
I either hope that gradle will reuse the classpath isolated worker (will need to test) or we can make a root level plugin to hold on to the various worker executor states.

Reply:
gradle will reuse the workers as much as possible (or so they say)